lang/cs/Org.Apache.REEF.Evaluator/Evaluator.cs

```diff
@@ -47,84 +43,43 @@ namespace Org.Apache.REEF.Evaluator
 {
     public sealed class Evaluator
     {
-        private static Logger logger = Logger.GetLogger(typeof(Evaluator));
-        private static int heartbeatPeriodInMs = Constants.DefaultEvaluatorHeartbeatPeriodInMs;
-        private static int heartbeatMaxRetry = Constants.DefaultEvaluatorHeartbeatMaxRetry;
-        private static IInjector injector;
-        private static EvaluatorConfigurations evaluatorConfig;
+        private static Logger _logger = Logger.GetLogger(typeof(Evaluator));
+
+        [Inject]
+        private Evaluator(
+            RuntimeClock clock,
```

Comment:
The original code was to inject RuntimeClock directly. There is no default for IClock. I could set default of bind RuntimeClock  to IClock but all the methods used in those classes for Clock are all defined in RuntimeClock rather than in IClock, example is Run(). we might want to visit it separately.